src/validation.cpp

```diff
@@ -353,6 +353,40 @@ bool CheckSequenceLocks(const CTransaction &tx, int flags, LockPoints* lp, bool
     return EvaluateSequenceLocks(index, lockPair);
 }
 
+bool GetUTXOCoin(const COutPoint& outpoint, Coin& coin, int height)
+{
+    LOCK(cs_main);
+    bool gotCoin = pcoinsTip->GetCoin(outpoint, coin);
+
+    if (height != chainActive.Tip()->nHeight) {
+        CSpentIndexKey key(outpoint.hash, outpoint.n);
+        CSpentIndexValue value;
+
+        if (GetSpentIndex(key, value)) {
+            if (value.blockHeight <= height) {
+                // Not Valid, already spent at this height
+                return false;
+            }
+        }
+        else
+        {
+            // Not spent, it was valid at this height
```

Comment:
My only concern with this call is that in this case, I am returning a default Coin (since it was later spent).  I doesn't affect the caller, but I don't know if this would be a problem/confusion later.

Reply:
i would remove this else case since it have no code but comment and i would combine if GetSpendIndex and value.blockHeight <= height to be in one if with &&